### PR TITLE
Cs udpsocket

### PR DESCRIPTION
**Pull request by shintiger**
I don't have experience on C#, this support remain many problem especially is memcpy issues.
At least this is a works version, I hope this give an insight for C# expert to optimize.

### CODE REVIEW

**Comment by RealyUniqueName** [comment]
@kLabz could you please check this PR?

**Comment by RealyUniqueName** [comment]
I'm not sure what to do with this one. We don't have proper tests for sockets atm.
@Aurel300 does new sys API cover udp sockets?

**Comment by Aurel300** [comment]
Yes, there are UDP sockets as well. Not a thorough test suite yet:

https://github.com/Aurel300/haxe-sys/blob/master/tests/test/TestUdp.hx

**Comment by Simn** [comment]
I'm not sure what to do with this one either. I prefer to solve this as part of the asys API, but that's still a while off so I'd be fine with this addition if it helps anyone. So if @shintiger is still interested in this, please update the PR and we can merge it.

File: std/cs/_std/sys/net/Socket.hx [hunk 1]
```diff
@@ -48,6 +49,13 @@ class Socket {
 
 	public var custom:Dynamic;
 
+	/**
+		Creates a new unconnected socket.
+	**/
+	public function new() : Void {
```
Comment (kLabz) [inline std/cs/_std/sys/net/Socket.hx]: Duplicated field; seems like the merge messed up with `function init()` below now named `new()`

**Pull request merged**